tsdb/chunks/chunks.go

```diff
@@ -308,7 +318,66 @@ func MergeChunks(a, b chunkenc.Chunk) (*chunkenc.XORChunk, error) {
 	if bit.Err() != nil {
 		return nil, bit.Err()
 	}
-	return newChunk, nil
+	newChks, err := breakLargeChunks(newChunk, MaxSamplesInChunk)
```

Comment:
I think what you did is essentially moved this function to inside the `MergeChunk`. 

What about actually checking the num samples while appending to the new chunk in `for aok && bok {` loop?

I think we can avoid building the same chunk twice (!)

Reply:
+1 on this, build 2 chunks in the fly.